src/Logs/Monolog/src/Handler.php

```diff
@@ -40,13 +51,40 @@ protected function write($record): void
             ->setSeverityNumber(API\Map\Psr3::severityNumber($record['level_name']))
             ->setSeverityText($record['level_name'])
             ->setBody($formatted['message'])
-            ->setAttribute('channel', $record['channel'])
         ;
-        foreach (['context', 'extra'] as $key) {
+        $attributes = [];
+        foreach (['extra', 'context'] as $key) {
             if (isset($formatted[$key]) && count($formatted[$key]) > 0) {
-                $logRecord->setAttribute($key, $formatted[$key]);
+                foreach ($formatted[$key] as $k => $v) {
+                    if ($mapped = self::mapToSemConv($k, $v, $record[$key][$k])) {
+                        $attributes += $mapped;
+                    } else {
+                        $attributes[$k] = $v;
+                    }
+                }
             }
         }
-        $this->logger->emit($logRecord);
+        $logRecord->setAttributes($attributes);
+        $this->getLogger($record['channel'])->emit($logRecord);
```

Comment:
Can you please confirm that this channel and logger lookup is fast enough that the performance impact is insignificant? I want to make sure the spec doesn't create a situation when we can't implement it in a performant way.

Reply:
I added a benchmark test, and ran it with the original code (one logger, fetched from constructor), and multiple loggers fetched/created on-demand. I see very no noticeable difference between the two.

This is the output from [phpbench](https://phpbench.readthedocs.io/) for the two tests:

Original:

+------+--------------+---------------+--------------------------+------+------------+---------------+--------------+----------------+
| iter | benchmark    | subject       | set                      | revs | mem_peak   | time_avg      | comp_z_value | comp_deviation |
+------+--------------+---------------+--------------------------+------+------------+---------------+--------------+----------------+
| 0    | LoggingBench | benchEmitLogs | 1 channel, 100 logs      | 10   | 2,761,912b | 1,589.400μs   | +0.00σ       | +0.00%         |
| 0    | LoggingBench | benchEmitLogs | 1 channel, 10000 logs    | 10   | 2,761,912b | 15,075.000μs  | +0.00σ       | +0.00%         |
| 0    | LoggingBench | benchEmitLogs | 4 channels, 100 logs     | 10   | 2,761,912b | 1,593.900μs   | +0.00σ       | +0.00%         |
| 0    | LoggingBench | benchEmitLogs | 4 channels, 10000 logs   | 10   | 2,761,920b | 150,492.200μs | +0.00σ       | +0.00%         |
| 0    | LoggingBench | benchEmitLogs | 16 channels, 100 logs    | 10   | 2,761,912b | 1,604.300μs   | +0.00σ       | +0.00%         |
| 0    | LoggingBench | benchEmitLogs | 16 channels, 10000 logs  | 10   | 2,761,920b | 148,999.000μs | +0.00σ       | +0.00%         |
| 0    | LoggingBench | benchEmitLogs | 256 channels, 100 logs   | 10   | 2,761,912b | 1,641.100μs   | +0.00σ       | +0.00%         |
| 0    | LoggingBench | benchEmitLogs | 256 channels, 10000 logs | 10   | 2,761,920b | 162,228.200μs | +0.00σ       | +0.00%         |
+------+--------------+---------------+--------------------------+------+------------+---------------+--------------+----------------+

Updated

+------+--------------+---------------+--------------------------+------+------------+---------------+--------------+----------------+
| iter | benchmark    | subject       | set                      | revs | mem_peak   | time_avg      | comp_z_value | comp_deviation |
+------+--------------+---------------+--------------------------+------+------------+---------------+--------------+----------------+
| 0    | LoggingBench | benchEmitLogs | 1 channel, 100 logs      | 10   | 2,761,912b | 1,585.200μs   | +0.00σ       | +0.00%         |
| 0    | LoggingBench | benchEmitLogs | 1 channel, 10000 logs    | 10   | 2,761,912b | 14,941.200μs  | +0.00σ       | +0.00%         |
| 0    | LoggingBench | benchEmitLogs | 4 channels, 100 logs     | 10   | 2,761,912b | 1,603.600μs   | +0.00σ       | +0.00%         |
| 0    | LoggingBench | benchEmitLogs | 4 channels, 10000 logs   | 10   | 2,761,920b | 155,136.500μs | +0.00σ       | +0.00%         |
| 0    | LoggingBench | benchEmitLogs | 16 channels, 100 logs    | 10   | 2,761,912b | 1,603.300μs   | +0.00σ       | +0.00%         |
| 0    | LoggingBench | benchEmitLogs | 16 channels, 10000 logs  | 10   | 2,761,920b | 155,178.900μs | +0.00σ       | +0.00%         |
| 0    | LoggingBench | benchEmitLogs | 256 channels, 100 logs   | 10   | 2,761,912b | 1,640.200μs   | +0.00σ       | +0.00%         |
| 0    | LoggingBench | benchEmitLogs | 256 channels, 10000 logs | 10   | 2,761,920b | 158,387.200μs | +0.00σ       | +0.00%         |
+------+--------------+---------------+--------------------------+------+------------+---------------+--------------+----------------+